packages/back-end/src/models/FeatureRevisionLogModel.ts

```diff
@@ -0,0 +1,79 @@
+import {
+  FeatureRevisionLogInterface,
+  featureRevisionLogValidator,
+} from "back-end/src/validators/feature-revision-log";
+import { MakeModelClass } from "./BaseModel";
+
+export const COLLECTION_NAME = "featurerevisionlog";
+
+const BaseClass = MakeModelClass({
+  schema: featureRevisionLogValidator,
+  collectionName: COLLECTION_NAME,
+  idPrefix: "frl_",
+  auditLog: {
+    entity: "featureRevisionLog",
+    createEvent: "featureRevisionLog.create",
+    updateEvent: "featureRevisionLog.update",
+    deleteEvent: "featureRevisionLog.delete",
+  },
+  globallyUniqueIds: true,
+  additionalIndexes: [
+    {
+      fields: {
+        featureId: 1,
+        version: 1,
+      },
+    },
+  ],
+});
+
+export class FeatureRevisionLogModel extends BaseClass {
+  protected canRead(doc: FeatureRevisionLogInterface): boolean {
+    const { feature } = this.getForeignRefs(doc);
+
+    return this.context.permissions.canReadSingleProjectResource(
+      feature?.project
+    );
+  }
+  protected canCreate(doc: FeatureRevisionLogInterface): boolean {
+    const { feature } = this.getForeignRefs(doc);
+    if (!feature) {
+      throw new Error("Feature not found for FeatureRevisionLog");
+    }
+    return (
+      this.context.permissions.canCreateFeature(feature) ||
+      this.context.permissions.canManageFeatureDrafts(feature)
+    );
+  }
+  protected canUpdate(existing: FeatureRevisionLogInterface): boolean {
```

Comment:
Do we ever want to allow updating or deleting a revision log?  That's kind of the point of a log like this, it's an audit log of all changes that happened.

I could see an argument for deleting when the parent feature gets deleted, but I can't think of a reason to ever allow updating.